File: Python/hamming.py
def distance(strand_a, strand_b):
    count = 0
    if len(strand_a) == len(strand_b):
        for x in range(len(strand_a)):
            if strand_a[x] != strand_b[x]:
                count += 1
    else:
        raise ValueError("Strands must be of equal length.")
    return count

File: Ruby/hamming.rb

class Hamming
  def self.compute(strand_a, strand_b)
    count = 0
    for index in (0..strand_a.length) do
      if strand_a[index] != strand_b[index] then
        count += 1
      end
    end
    return count
  end
end

  puts Hamming.compute('GGACTGAAATCTG', 'GGACTGAAATCTG')

  puts Hamming.compute('GGACGGATTCTG', 'AGGACGGATTCT')